src/dataflow/utilities/node.ts

```diff
@@ -240,3 +240,54 @@ export const NodePlotColors = ["#0592AF", "#ebb93e", "#eb813e", "#df4a4a", "#e44
                                "#c60e0e", "#cc0860", "#b81da1", "#8d27ad", "#8b989f",
                                "#5dd581", "#3cc8f5", "#aeb9bf", "#92e3aa", "#7ad9f8",
                                "#5d6e77", "#31bc5a", "#0caadd"];
+
+export interface ProgramRunTime {
+  text: string;
+  val: number;
+}
+export const ProgramRunTimes: ProgramRunTime[] = [
+  {
+    text: "30 Seconds",
+    val: 30
+  },
+  {
+    text: "1 Minute",
+    val: 60
+  },
+  {
+    text: "5 Minutes",
+    val: 300
+  },
+  {
+    text: "10 Minutes",
+    val: 600
+  },
+  {
+    text: "15 Minutes",
+    val: 900
+  },
+  {
+    text: "30 Minutes",
+    val: 1800
+  },
+  {
+    text: "1 Hour",
+    val: 3600
+  },
+  {
+    text: "2 Hours",
+    val: 7200
+  },
+  {
+    text: "4 Hours",
+    val: 14400
+  },
+  {
+    text: "8 Hours",
+    val: 28800
+  },
+  {
+    text: "Until Stopped",
+    val: Infinity
+  },
+];
```

Comment:
The idea here is simple.  Give the user options to run the program for some set time rather than always running the program until it is manually stopped (which caused issues in DF2 as students would not remember to stop a program).  No clear answer on exactly what these values will be (or if we're planning to allow the user to have total flexibility when entering a run time - ala "run for 3 minutes and 37 seconds") but this is an initial estimate based on the recommendations that Lisa and Colin gave to me a few weeks back when they initially pitched this feature.

Reply:
I don't think this affects the PR, but I had a brief conversation with Sherry about this. I don't think we should offer "Until Stopped" as an option. We should offer the longest option that makes sense -- a week? a month? a year? -- but we should not imply that we're willing to keep going until the end of time.

Reply:
Ah, glad to hear that.  The run forever option caused tremendous grief in DF2.  I'm going to remove the "Until Stopped" option here.  I'll ping the inspect channel for what these intervals should be.